persistence.js

```diff
@@ -287,31 +49,20 @@ class MongoPersistence extends CachedPersistence {
       return
     }
 
-    let published = 0
-    let errored = false
-    const bulk = this._cl.subscriptions.initializeOrderedBulkOp()
-    subs
-      .forEach((sub) => {
-        const subscription = Object.assign({}, sub)
-        subscription.clientId = client.id
-        bulk.find({
-          clientId: client.id,
-          topic: sub.topic
-        }).upsert().updateOne({
-          $set: decorateSubscription(subscription, this._opts)
-        })
+    const addSubs1 = this.asyncPersistence.addSubscriptions(client, subs)
+    // promisify
+    const addSubs2 = new Promise((resolve, reject) => {
+      this._addedSubscriptions(client, subs, (err) => {
+        if (err) {
+          reject(err)
+        } else {
+          resolve()
+        }
```

Comment:
this is wrong as you should ensure the first completes before the second one

Reply:
I fixed this one and removeSubscriptions but test/own.js still fails and test/abs.js still succeeds.